### DISCUSSION TITLE
Non-english words in brace label

### DISCUSSION THREAD

**Discussion by sindzicat** [ManimCommunity/manim#3071] [Q&A]
Hello!

I'm learning Manim and has stuck with this example: https://docs.manim.community/en/stable/examples.html#braceannotation

I need to replace `"Horizontal distance"` to `"Горизонтальное смещение"`. I found, that brace label uses TeX. I know how to produce text on Russian using XeLaTeX:

```tex
\documentclass{article}

\usepackage{amsmath,amsthm,amssymb,amsfonts}
\usepackage{color}

\usepackage{polyglossia}
\setmainlanguage{russian}
\setotherlanguage{english}


\usepackage{fontspec}
\usepackage{unicode-math}

\setmainfont{XITS}
\setmathfont{CMU Classical Serif}

\pagestyle{empty}

\begin{document}

Test equation (тестовое уравнение): $\sum_{n=0}^\infty F_{\text{упр.}n}$

\end{document}
```

But I don't know how to make brace label to use my preamble and XeLaTeX. Is this possible just use simple text in brace preamble (without TeX)?

**Comment by MrDiver** [comment]
You can use the `TexTemplate` module for this. Just use one of the predefined templates in the [`TexTemplateLibrary`](https://docs.manim.community/en/stable/reference/manim.utils.tex_templates.TexTemplateLibrary.html?highlight=TexTemplateLibrary) and modify it to your liking or define your own with [`TexTemplate`](https://docs.manim.community/en/stable/reference/manim.utils.tex.TexTemplate.html?highlight=TexTemplate)
Currently i don't know if you can get `xelatex` to coorporate with manim because we need dvi files as output. But in general that would be the way to go about it.

**Reply by sindzicat** [reply]
Sorry for a long silence. I knew about `TexTemplate`, and my problem was that `get_tex()` function don't provide a way to use `TexTemplate`. At least I didn't find a way to use `get_tex()` with `TexTemplate`. But I found a workaround. I found that [source code](https://github.com/ManimCommunity/manim/blob/95bca6f24b42bd8bfec59f20d3a26457372f9d83/manim/mobject/svg/brace.py#L141) of `get_tex()` function is very simple. It just [use](https://github.com/ManimCommunity/manim/blob/95bca6f24b42bd8bfec59f20d3a26457372f9d83/manim/mobject/svg/brace.py#L126) `put_at_tip()` that [uses](https://github.com/ManimCommunity/manim/blob/95bca6f24b42bd8bfec59f20d3a26457372f9d83/manim/mobject/svg/brace.py#L128) `next_to()` function.

So unstead of `get_tex()` function I just used `next_to()` function with my own `TexTemplate()`:

```python
%%manim -qm -v WARNING BraceAnnotationRu

PREAMBLE = r'''
\usepackage{amsmath,amsthm,amssymb,amsfonts}
\usepackage{color}

\usepackage{polyglossia}
\setmainlanguage{russian}
\setotherlanguage{english}


\usepackage{fontspec}
\usepackage{unicode-math}

\setmainfont{XITS}
\setmathfont{CMU Classical Serif}
'''

class BraceAnnotationRu(Scene):
    def construct(self):
        dot = Dot([-2, -1, 0])

        dot2 = Dot([2, 1, 0])

        line = Line(dot.get_center(), dot2.get_center()).set_color(ORANGE)

        b1 = Brace(line)

        b1text = b1.get_text("Horizontal distance")

        b2 = Brace(line, direction=line.copy().rotate(PI / 2).get_unit_vector())

        tex_tpl = TexTemplate(
            tex_compiler='xelatex',
            output_format='.xdv',
            preamble=PREAMBLE
        )

        tip_2 = MathTex(r'\vec F_\text{абс}', tex_template=tex_tpl)

        self.add(line, dot, dot2, b1, b2, b1text, tip_2.next_to(b2.get_tip(), np.round(b2.get_direction())))
```

![image](https://user-images.githubusercontent.com/33900895/215403290-4207d309-28e4-4fba-ad87-342e71f1355e.png)
